DMTN-306.tex

```diff
@@ -1,79 +1,135 @@
+
+Rubin has developed the tool \texttt{rucio\_register}\footnote{https://github.com/lsst/rucio\_register}, which allows for the selection of existing files from a Butler repository based on specified criteria. The tool attaches Rubin-specific metadata to these files and registers them into one or more Rucio Datasets. The metadata, encoded as a JSON record, contains a minimal set of information extracted from the origin Butler repository. This ensures that replicated files can be ingested properly into the local Butler repository at the destination facility.
+
+Only files that require replication to another facility are registered with Rucio. As a result, Rubin’s instance of Rucio is aware only of files replicated across processing facilities. Files that are local to each facility and not subject to replication remain known only to that facility's Butler and are not registered in Rucio.
+
+\subsection{Ingestion at reception}
+\label{ingestion}
+
+FTS notifies Rucio about the completion of individual file transfers. Rubin's \texttt{ctrl\_rucio\_ingest}\footnote{https://github.com/lsst-dm/ctrl\_rucio\_ingest}, an extension to Rucio's Hermes daemon, uses Kafka as a mechanism to signal the destination Rubin facility that a new file was replicated and to take appropriate actions.
+
+Messages distributed through Rubin's Kafka control-plane include Rubin-specific metadata. Those messages are received by Rubin's \texttt{ctrl\_ingestd}\footnote{https://github.com/lsst-dm/ctrl\_ingestd}, a daemon running at each destination facility responsible for ingesting newly replicated files into the local Butler repository.
 
-% ADD CONTENT HERE
-% You can also use the \input command to include several content files.
+Each facility only receives notifications about files successfully replicated to the storage endpoints it operates. This is achieved by following a simple convention: the name of Kafka topic the notification is sent to is named after the name of the Rucio storage element. Each facility's \texttt{ctrl\_ingestd} is configured to only monitor Kafka messages specifically targeted to the facility's RSEs (see figure \ref{fig:kafka-control-plane}).
 
-\appendix
-% Include all the relevant bib files.
-% https://lsst-texmf.lsst.io/lsstdoc.html#bibliographies
-\section{References} \label{sec:bib}
-\renewcommand{\refname}{} % Suppress default Bibliography section
-\bibliography{local,lsst,lsst-dm,refs_ads,refs,books}
+\begin{figure}[h]
+\includegraphics[width=0.9\textwidth, center]{images/Hermes and Ingestd.png}
+\caption{Notifications about successful file transfers are emitted by \texttt{ctrl\_rucio\_ingest} using Kafka topics named after the destination RSE. At the receiving facility \texttt{ctrl\_ingestd} monitors those notifications and ingests the newly received file into the local Butler repository. The JSON-encoded, Rubin-specific metadata associated to the file when it was first registered into Rucio contains the details needed for ingestion.}
```

Comment:
Notifications about successful file transfers are emitted by \texttt{HermesK} using Kafka topics named after the destination RSE. At the receiving facility \texttt{ingestd} monitors those notifications and ingests the newly received file into the local Butler repository. The JSON-encoded, Rubin-specific metadata associated to the file when it was first registered into Rucio contains the details needed for ingestion.}

Reply:
Implemented